wordpress-mobile/WordPress-Android
# [JP Content Migration Flow] Implement success card dialog

**Reviewer**: this is a huge PR, but about 400 new lines are unit tests. If you prefer that I split the tests into a separate PR, please let me know. 
The tests I've added don't cover many important parts of the flow, so more tests should be added later.

Fixes #17479

<img width="354" alt="image" src="https://user-images.githubusercontent.com/14964993/203753703-b4a966ee-3691-4dd6-90f2-ca5e0a4a33d3.png">
<img width="355" alt="image" src="https://user-images.githubusercontent.com/14964993/203753143-f091f597-27a3-4951-9ce0-38e63b03b3ba.png">

To test:
1 - Install WordPress;
2 - Install Jetpack using the same variant (e.g. if installed WordPress version was `jalapenoDebug`, install Jetpack `jalapenoDebug`);
3 - The migration success card **SHOULD NOT** appear on "My Site" ("HOME" or "MENU");
4 - Set the value of `isMigrationCompleted` to `true` (`org/wordpress/android/ui/mysite/MySiteViewModel.kt:438`);
5 - Build and install Jetpack again (like step 2);
6 - The migration success card **SHOULD** appear on "My Site" (on "HOME" **and** "MENU");
7 - Tap on migration card. Migration screen should be opened as seen in the screenshots.
8 - Tap on "Need help?" button. It should open Help/Support screen.
9 - Tap the back button.
10 - Tap on "Got it". Main screen should be seen again.

It's worth also double-checking the data migration flow. This PR should not impact the existing migration steps.

When you follow the standard migration flow, the success card should appear without step 4 since [we're updating the migration success flag in the end of the flow](https://github.com/wordpress-mobile/WordPress-Android/pull/17527/files#diff-1497c9055b16400d07a9852b373d377024d5cbb1d36f0822069526579e2f6de6R85).

## Regression Notes
1. Potential unintended areas of impact
Existing migration flow screens.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and added unit tests.

3. What automated tests I added (or what prevented me from doing so)
`JetpackMigrationViewModelTest.kt`

PR submission checklist:
